src/ring_theory/finiteness.lean

```diff
@@ -35,7 +35,8 @@ variables [add_comm_group M] [module R M]
 variables [add_comm_group N] [module R N]
 
 /-- A module over a commutative ring is `finite` if it is finitely generated as a module. -/
-class module.finite : Prop := (out : (⊤ : submodule R M).fg)
+class module.finite (R : Type*) (M : Type*) [comm_semiring R] [add_comm_monoid M] [module R M] :
```

Comment:
This is just to be able to state `iff_add_monoid_fg`. I didn't generalize anything else to `comm_semiring`.